flow/workflows/cancel_table_addition_flow.go

```diff
@@ -0,0 +1,221 @@
+		StartToCloseTimeout: time.Minute * 15,
+		HeartbeatTimeout:    2 * time.Minute,
+		RetryPolicy: &temporal.RetryPolicy{
+			InitialInterval:    time.Second * 10,
+			BackoffCoefficient: 2.0,
+			MaximumInterval:    time.Minute * 5,
+		},
+	}
+	cleanupIncompleteTablesCtx := workflow.WithActivityOptions(ctx, cleanupIncompleteTablesOptions)
+
+	err = workflow.ExecuteActivity(cleanupIncompleteTablesCtx, cancelTableAddition.CleanupIncompleteTablesInStats,
+		flowJobName, snapshottedTables).Get(ctx, nil)
+	if err != nil {
+		logger.Error("Failed to cleanup incomplete tables in stats", "error", err)
+		return err
+	}
+	logger.Info("Successfully cleaned up incomplete tables in stats", "flowName", flowJobName)
+
+	getFlowConfigOptions := workflow.ActivityOptions{
+		StartToCloseTimeout: time.Minute * 5,
+		RetryPolicy: &temporal.RetryPolicy{
+			InitialInterval:    time.Second * 5,
+			BackoffCoefficient: 2.0,
+			MaximumInterval:    time.Minute * 1,
+			MaximumAttempts:    5,
+		},
+	}
+	getFlowConfigCtx := workflow.WithActivityOptions(ctx, getFlowConfigOptions)
+
+	var flowConfig *protos.FlowConnectionConfigs
+	err = workflow.ExecuteActivity(getFlowConfigCtx, cancelTableAddition.GetFlowConfigFromCatalog, flowJobName).Get(ctx, &flowConfig)
+	if err != nil {
+		logger.Error("Failed to get flow config from catalog", "error", err)
+		return err
+	}
+	logger.Info("Retrieved flow config", "flowName", flowJobName, "tableCount", len(flowConfig.TableMappings))
+
+	// update table mappings for upcoming request
+	flowConfig.TableMappings = snapshottedTables
+	flowConfig.DoInitialSnapshot = false
+	coreConfig := proto_conversions.FlowConnectionConfigsToCore(flowConfig, 0)
+
+	state := NewCDCFlowWorkflowState(ctx, logger, coreConfig)
+	// update table OIDs for upcoming request
+	if len(tableOIDs) > 0 {
+		state.SyncFlowOptions.SrcTableIdNameMapping = tableOIDs
+		logger.Info("Set source table ID name mapping in state",
+			"flowName", flowJobName, "mappingCount", len(tableOIDs))
+	}
+
+	// this allows us to skip setup and snapshot
+	state.CurrentFlowStatus = protos.FlowStatus_STATUS_RUNNING
+
+	cleanupCurrentParentMirrorOptions := workflow.ActivityOptions{
+		StartToCloseTimeout: time.Minute * 15,
+		HeartbeatTimeout:    2 * time.Minute,
+		RetryPolicy: &temporal.RetryPolicy{
+			InitialInterval:    time.Second * 5,
+			BackoffCoefficient: 2.0,
+			MaximumInterval:    time.Minute * 1,
+		},
+	}
+	cleanupCurrentParentMirrorCtx := workflow.WithActivityOptions(ctx, cleanupCurrentParentMirrorOptions)
+
+	err = workflow.ExecuteActivity(cleanupCurrentParentMirrorCtx, cancelTableAddition.CleanupCurrentParentMirror, flowJobName).Get(ctx, nil)
+	if err != nil {
+		logger.Error("Failed to cleanup current parent mirror", "error", err)
+		return err
+	}
+
+	logger.Info("Successfully cleaned up current parent mirror workflow", "flowName", flowJobName)
+
+	createCdcJobEntryOptions := workflow.ActivityOptions{
+		StartToCloseTimeout: time.Minute * 3,
```

Comment:
```suggestion
		StartToCloseTimeout: time.Minute * 5,
```
splitting hairs tbh but the other PG reads are 5-15